src/App.js

```diff
@@ -1,26 +1,114 @@
 import React from 'react';
-import logo from './logo.svg';
+import Bucket from './Bucket';
 import './App.css';
+import { Container } from 'reactstrap';
+import {
+  InputGroup,
+  InputGroupAddon,
+  InputGroupText,
+  Col,
+  Row,
+  Label,
+  Input
+} from 'reactstrap';
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+class App extends React.Component {
+  state = {
+    income: 0,
+    buckets: {
+      "Fixed Costs": 50,
+      Investments: 10,
+      Savings: 20,
+      "Guilt-Free Spending": 20,
+    }
+  }
+
+  updateIncome = e => {
+    this.setState({ income: e.target.value });
+  };
+
+  percentUpdated = (bucketName, percentage) => {
+    let buckets = this.state.buckets;
```

Comment:
```suggestion
    const buckets = this.state.buckets;
```

Nit but this can be a constant.

Reply:
thanks!